internals/src/lib.rs

```diff
@@ -35,6 +35,12 @@ pub mod rust_version {
     include!(concat!(env!("OUT_DIR"), "/rust_version.rs"));
 }
 
+#[doc(hidden)]
+pub mod _export {
+    #[cfg(feature = "alloc")]
+    pub extern crate alloc;
```

Comment:
In e8a42d5851000d210ea9b886478d83f3e7c077a6:

Just commenting to note that I did think about what happens when `alloc` gets enabled on `internals` but not on the calling crate. I believe that, since all the generated code only accesses `alloc` through `$crate::_export::alloc`, that everything will continue to compile, though the calling crates' types may get some extra methods.

Reply:
Yes, and the calling crates getting extra methods is fully intended. (In principle we need to be careful about not adding same-named methods but CI will trivially catch that as long as it tests the feature.)